runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -7022,6 +7054,24 @@ void nic_amo_nf_buff_flush(void) {
   rwlock_unlock(&amo_nf_buff_global_info.lock);
 }
 
+// Flush buffered atomic operations for this task
+static
+inline
+void nic_amo_nf_buff_task_flush(void) {
+  if (chpl_task_canMigrateThreads()) {
+    nic_amo_nf_buff_flush();
+  } else {
+    amo_nf_buff_thread_info_t* info = &amo_nf_buff_thread_info;
+    // Safe to check inited/vi outside the lock because no other thread can be
+    // modifying them, but concurrent flushing from other threads is possible.
+    if (info->inited && info->vi > 0) {
+      spinlock_lock(&info->lock);
+      flush_amo_nf_buff(info);
```

Comment:
Similar to the other case ...

Reply:
`flush_amo_nf_buff` also checks `info->vi> 0`